lib/msal-node/src/client/PublicClientApplication.ts

```diff
@@ -74,4 +84,83 @@ export class PublicClientApplication extends ClientApplication implements IPubli
+        const { openBrowser, successTemplate, errorTemplate, ...remainingProperties } = request;
+
+        return new Promise(async (resolve, reject) => {
+            const server = createServer((req: IncomingMessage, res: ServerResponse) => {
+                const url = req.url;
+                if (!url) {
+                    res.end(errorTemplate || "Error occurred loading redirectUrl");
+                    return;
+                } else if (url === CommonConstants.FORWARD_SLASH) {
+                    res.end(successTemplate || "Auth code was successfully acquired. You can close this window now.");
+                    return;
+                }
+
+                const authCodeResponse = UrlString.getDeserializedQueryString(url);
+                const code = authCodeResponse.code;
+                if (code) {
+                    const redirectUri = `${Constants.HTTP_PROTOCOL}${req.headers.host}`;
+                    res.writeHead(302, { location: redirectUri }); // Prevent auth code from being saved in the browser history
+                    res.end();
+
+                    if (!req.headers.host) {
+                        server.close();
+                        reject(NodeAuthError.createHostHeaderMissingError());
+                    }
+                    
+                    const clientInfo = authCodeResponse.client_info;
+                    const tokenRequest: AuthorizationCodeRequest = {
+                        code: code,
+                        scopes: OIDC_DEFAULT_SCOPES,
+                        redirectUri: redirectUri,
+                        codeVerifier: verifier,
+                        clientInfo: clientInfo || CommonConstants.EMPTY_STRING
+                    };
+                    this.acquireTokenByCode(tokenRequest).then(response => {
+                        resolve(response);
+                    }).catch((e) => {
+                        reject(e);
+                    }).finally(() => {
+                        server.close();
+                    });
+                } else if (authCodeResponse.error) {
+                    res.end(errorTemplate || `Error occurred: ${authCodeResponse.error}`);
+                } else {
+                    res.end(errorTemplate || "Unknown error occurred");
+                }
+            });
+            server.listen(0); // Listen on any available port
+            const port: number = await new Promise((resolvePort) => {
+                const id = setInterval(() => {
+                    const address = server.address();
+                    if (typeof address === "string") {
+                        clearInterval(id);
+                        reject(NodeAuthError.createAddressWrongTypeError());
+                    } else if (address && address.port) {
+                        clearInterval(id);
+                        resolvePort(address.port);
+                    }
+                }, 100);
```

Comment:
Can you create a constant for this value? And do we want this to be configurable?

Reply:
Created a constant. Don't think this needs to be configurable but we can re-evaluate if someone asks for it